components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java

```diff
@@ -55,33 +57,90 @@
     @Mock
     Job job;
 
+    @Mock
+    Job dynamicMockjob;
+
     // Camel fixtures
 
     @EndpointInject(uri = "mock:test")
     MockEndpoint mockEndpoint;
 
+    @EndpointInject(uri = "mock:error")
+    MockEndpoint errorEndpoint;
+
+
     @Override
     protected RouteBuilder createRouteBuilder() throws Exception {
         return new RouteBuilder() {
             @Override
             public void configure() throws Exception {
                 from("direct:start").to("spring-batch:mockJob").to("mock:test");
+                from("direct:dynamic").
+                        to("spring-batch:fake?jobFromHeader=true").
+                        errorHandler(deadLetterChannel("mock:error")).
+                        to("mock:test");
             }
         };
     }
 
+
+
     @Override
     public JndiRegistry createRegistry() throws Exception {
         JndiRegistry registry = super.createRegistry();
         registry.bind("jobLauncher", jobLauncher);
         registry.bind("alternativeJobLauncher", alternativeJobLauncher);
         registry.bind("mockJob", job);
+        registry.bind("dynamicMockjob", dynamicMockjob);
         return registry;
     }
 
     // Tests
 
     @Test
+    public void dynamicJobFailsIfHeaderNotPressent() throws Exception {
+
+        mockEndpoint.expectedMessageCount(0);
+        errorEndpoint.expectedMessageCount(1);
+
+        //dynamic job should fail as header is not present and the job is dynamic
```

Comment:
redundant comment, the test name is enough descriptive (btw. missing space after //)